reana_job_controller/schemas.py

```diff
@@ -42,6 +47,38 @@ class JobRequest(Schema):
     voms_proxy = fields.Bool(required=False)
     kubernetes_uid = fields.Int(required=False)
     kubernetes_memory_limit = fields.Str(required=False)
+    kubernetes_job_timeout = fields.Int(required=False, strict=True)
     unpacked_img = fields.Bool(required=False)
     htcondor_max_runtime = fields.Str(required=False)
     htcondor_accounting_group = fields.Str(required=False)
+
+    @validates("kubernetes_job_timeout")
+    def validate_kubernetes_job_timeout(self, value):  # noqa: D102
```

Comment:
Can't we perform the validation directly at the schema declaration level? (untested)

```python
...
kubernetes_job_timeout = fields.Int(strict=True, default=REANA_KUBERNETES_JOBS_TIMEOUT_LIMIT, validate=validate.Range(min=0, max=int(REANA_KUBERNETES_JOBS_MAX_USER_TIMEOUT_LIMIT)))
...
```

Reply:
If `REANA_KUBERNETES_JOBS_MAX_USER_TIMEOUT_LIMIT` is not an integer string like "10" but something like "10s", `int()` will fail.

Reply:
But I don't remember if the `default` value will be validated. Let me check.

Reply:
But it should fail, right? I don't think we're accepting "s" at the moment. If we want to accept them then the filed should be `Str` instead of `Int` in my opinion.

Reply:
> But it should fail, right?

If you are talking about `int(REANA_KUBERNETES_JOBS_MAX_USER_TIMEOUT_LIMIT)` then yes it will fail, but not in a nice way. This will make the job-controller crash. 

> If we want to accept them then the filed should be Str instead of Int in my opinion.

We want `Int` at the end, cause seconds is the only value that k8s accepts. [pre_load](https://marshmallow.readthedocs.io/en/stable/extending.html#pre-post-processor-invocation-order) comes before schema validation (where `Int` is checked) so if you enter a string it will be fine until `validation_schema` or `validation` steps are reached.